### ISSUE TITLE
[Bug]: yuv_to_rgb is not the inverse of rgb_to_yuv — round-trip error exceeds the test tolerance on ~13% of runs

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4044]
### What happened?

`kornia.color.yuv_to_rgb` is not the inverse of `kornia.color.rgb_to_yuv`. The two functions hardcode independently rounded matrices, and the product `K_yuv2rgb @ K_rgb2yuv` is off the identity by up to **1.36e-3** in the blue channel — larger than the `atol=1e-3` that `tests/color/test_yuv.py::TestRgbToYuv::test_round_trip_rgb_yuv_rgb` asserts. Since that test draws unseeded `torch.rand` input, it fails whenever one of its 60 pixels lands in the bad corner of the cube.

Measured on `main` (`35555b55`), the same test invocation repeated:

```
3 / 20 runs fail       # via pytest, unseeded
26 / 200 seeds fail    # torch.manual_seed(0..199), float32
27 / 200 seeds fail    # same seeds, float64
```

Every single failure is in the **B** channel, and it is not a float32 precision effect — float64 fails at the same rate, because the error is in the constants, not in the arithmetic.

This is a **library defect that happens to surface as a flaky test**, not a flaky test. Any user round-tripping RGB → YUV → RGB silently loses up to 1.4e-3 of blue.

#### Root cause

`kornia/color/yuv.py:56` (forward, BT.470-5 M/PAL) and `kornia/color/yuv.py:164` (inverse):

```python
# rgb_to_yuv
[[ 0.299,  0.587,  0.114],
 [-0.147, -0.289,  0.436],
 [ 0.615, -0.515, -0.100]]

# yuv_to_rgb
[[1.0,  0.0,    1.14 ],
 [1.0, -0.396, -0.581],
 [1.0,  2.029,  0.0  ]]
```

The inverse matrix is not the inverse of the forward matrix — it is a *separately rounded* copy of the published constants, and one literal is rounded badly. The exact inverse of the forward kernel is:

```
[[1.0, -0.00003946,  1.13982797],
 [1.0, -0.39461016, -0.58050032],
 [1.0,  2.03199968, -0.00048138]]
```

so `2.029` should be `2.032` (the published value is `2.03211`). With `U ∈ [-0.436, 0.436]`, that single 3.1e-3 coefficient error contributes `0.00311 × 0.436 = 1.36e-3` to B — exactly the observed worst case.

`K_yuv2rgb @ K_rgb2yuv - I`:

```
[[ 0.000100, -0.000100,  0.000000],
 [-0.000103,  0.000659, -0.000556],
 [ 0.000737,  0.000619, -0.001356]]
```

Worst-case round-trip error over the whole `[0, 1]³` RGB cube, and what each candidate kernel would give:

| `yuv_to_rgb` kernel | worst R | worst G | worst B |
|---|---|---|---|
| current | 1.0e-4 | 6.6e-4 | **1.36e-3** |
| current with only `2.029 → 2.032` | 1.0e-4 | 6.6e-4 | 3.0e-4 |
| published BT.470 inverse (`1.13983, -0.39465, -0.58060, 2.03211`) | 1.7e-5 | 6.3e-5 | 2.8e-4 |
| exact inverse of kornia's forward kernel | 0 | 0 | 0 |

The failing pixels are exactly where the analysis predicts — high R, high G, near-zero B, where the `rtol` term adds nothing to the `atol=1e-3` budget:

```
seed=4  pixel rgb=[0.5949, 0.9622, 0.0009]  |err_B| = 1.03e-3
seed=9  pixel rgb=[0.8091, 0.9533, 0.0172]  |err_B| = 1.16e-3
```

#### How it got here

#3535 asked to tighten the round-trip test, which had been running with `low_tolerance=True`. #3539 closed it by rewriting the test with an explicit `atol=1e-3` — but the underlying kernel mismatch of `1.36e-3` was never fixed, so the tolerance was set just *inside* the systematic error rather than outside it. The test has been intermittently red ever since.

### Minimal reproduction

```python
import torch
import kornia

K = torch.tensor([[0.299, 0.587, 0.114],
                  [-0.147, -0.289, 0.436],
                  [0.615, -0.515, -0.100]], dtype=torch.float64)
Kinv = torch.tensor([[1.0, 0.0, 1.14],
                     [1.0, -0.396, -0.581],
                     [1.0, 2.029, 0.0]], dtype=torch.float64)
print(Kinv @ K - torch.eye(3, dtype=torch.float64))
# tensor([[ 0.000100, -0.000100,  0.000000],
#         [-0.000103,  0.000659, -0.000556],
#         [ 0.000737,  0.000619, -0.001356]])   <- B channel, > the test's atol=1e-3

# worst case reached by an actual image, deterministically:
rgb = torch.tensor([1.0, 1.0, 0.0], dtype=torch.float64).view(1, 3, 1, 1)
back = kornia.color.yuv_to_rgb(kornia.color.rgb_to_yuv(rgb))
print((back - rgb).abs().max().item())   # 0.001356...

# and the flake itself:
for seed in (4, 9):
    torch.manual_seed(seed)
    rgb = torch.rand(3, 4, 5)
    back = kornia.color.yuv_to_rgb(kornia.color.rgb_to_yuv(rgb))
    print(seed, (back - rgb).abs().max().item())   # 0.001033, 0.001163  > atol=1e-3
```

### Environment

- Kornia version: 0.9.0rc1 (`main` @ `35555b55`)
- PyTorch version: 2.9.1
- Python version: 3.11.14
- OS / device (CPU, CUDA, ROCm, MPS): macOS-26.5.1-arm64, CPU

Not environment-specific: the defect is in the hardcoded constants, and it reproduces identically at float32 and float64.

### Anything else?

**Suggested fix.** Replace the hardcoded `yuv_to_rgb` matrix with the exact inverse of the `rgb_to_yuv` matrix (as literals, so the op stays JIT/`torch.compile`-friendly and allocation-free), then tighten the round-trip test to something the math actually supports and seed it. That is a real output change for `yuv_to_rgb`, `yuv420_to_rgb` and `yuv422_to_rgb` and needs a changelog entry — bounded, over the documented YUV domain (`Y ∈ [0,1]`, `U ∈ [-0.436, 0.436]`, `V ∈ [-0.615, 0.615]`), by:

| channel | max change |
|---|---|
| R | 1.2e-4 |
| G | 9.1e-4 |
| B | 1.6e-3 |

An alternative — loosening the test tolerance to 2e-3 — would make CI green while leaving `yuv_to_rgb` a non-inverse. The constants are wrong; the tolerance is only the messenger.

**Related coverage gap found while diagnosing this.** #3539 also deleted five of the six test classes in `tests/color/test_yuv.py` (494 lines removed, 53 added), leaving only `TestRgbToYuv`. As of `main`, `rgb_to_yuv420`, `rgb_to_yuv422`, `yuv420_to_rgb` and `yuv422_to_rgb` have **no tests at all** (`grep -rn "yuv420_to_rgb\|yuv422_to_rgb" tests/` hits only `tests/api_surface.json`), and `yuv_to_rgb` is exercised only as the second half of this one flaky round-trip. Whoever fixes the kernel will need that coverage back to land the change safely; it probably deserves its own issue.

---

Posted on behalf of [@ducha-aiki](https://github.com/ducha-aiki) by Claude (Opus 5).

**Comment by ducha-aiki** [comment]
Filed the coverage gap noted at the end of this issue as #4045: #3539 deleted five of the six test classes in `tests/color/test_yuv.py` (494 lines out, 53 in), so `rgb_to_yuv420`, `rgb_to_yuv422`, `yuv420_to_rgb` and `yuv422_to_rgb` currently have no tests at all, and `yuv_to_rgb` only the one line of the round-trip described above.

That matters for fixing this issue: the kernel correction changes `yuv_to_rgb`, `yuv420_to_rgb` and `yuv422_to_rgb` output, and there is nothing in the suite that would catch a mistake in it. Three semantic mutations of the subsampled functions pass `tests/color/` unchanged today. Best landed together.

---

Posted on behalf of [@ducha-aiki](https://github.com/ducha-aiki) by Claude (Opus 5).
